src/components/common/Members/filterMembers.ts

```diff
@@ -1,55 +1,63 @@
 import { FormValues } from '~common/ColonyMembers/MembersFilter';
 import { VerificationType } from '~common/ColonyMembers/MembersFilter/filtersConfig';
 
-import { Contributor, Watcher } from '~types';
+import { Address, Contributor, Watcher } from '~types';
+import { isUserVerified } from '~utils/verifiedUsers';
 
 const filterMemberBySearchTerm = (
   member: Contributor | Watcher,
-  searchTerm?: string,
+  searchTerm: string,
 ) => {
   return (
-    searchTerm &&
-    (member?.user?.profile?.displayName
+    member?.user?.profile?.displayName
       ?.toLowerCase()
       .includes(searchTerm.toLowerCase()) ||
-      member?.user?.walletAddress
-        ?.toLowerCase()
-        .includes(searchTerm.toLowerCase()) ||
-      member?.user?.name.toLowerCase().includes(searchTerm.toLowerCase()))
+    member?.address?.toLowerCase().includes(searchTerm.toLowerCase()) ||
+    member?.user?.name.toLowerCase().includes(searchTerm.toLowerCase())
   );
 };
 
 const filterMemberByVerificationStatus = (
   member: Contributor | Watcher,
+  whitelistedAddresses: Address[],
   verificationType?: VerificationType,
 ) => {
+  const isVerified = isUserVerified(member.address, whitelistedAddresses);
+
   if (verificationType === VerificationType.Verified) {
-    return true;
-    // return member.isWhitelisted;
+    return isVerified;
   }
 
   if (verificationType === VerificationType.Unverified) {
-    return true;
-    // return !member.isWhitelisted;
+    return !isVerified;
   }
 
   return true;
-  // return !member.isWhitelisted;
 };
 
-export const filterMembers = <M extends Contributor | Watcher>(
-  members: M[],
+export const filterMembers = <Member extends Contributor | Watcher>(
+  members: Member[],
+  whitelistedAddresses: Address[],
   searchTerm?: string,
   filters?: FormValues,
-): M[] => {
-  /* No filters */
-  if (!searchTerm && filters?.verificationType === VerificationType.All) {
-    return members;
+): Member[] => {
+  let filtered = members;
+
+  if (filters?.verificationType !== VerificationType.All) {
```

Comment:
There's [an issue](https://github.com/JoinColony/colonyCDapp/issues/276) to move this to the members lambda. Have not addressed here.